Doc/whatsnew/3.8.rst

```diff
@@ -90,10 +90,20 @@ New Modules
 Improved Modules
 ================
 
-
 Optimizations
 =============
 
+* :func:`shutil.copyfile`, :func:`shutil.copy`, :func:`shutil.copy2`,
+  :func:`shutil.copytree` and :func:`shutil.move` use platform specific
+  "zero-copy" syscalls such as :func:`os.sendfile` in order to copy the file
+  more efficiently.
+  "zero-copy" means that the copying operation occurs within the kernel,
+  avoiding the use of userspace buffers as in "``outfd.write(infd.read())``".
```

Comment:
As with the main docs, glossing over the details is likely a better option than trying to come up with a cross-platform summary of the semantics of kernel level copy operations.